Insights/Networking/WebService.swift

```diff
@@ -57,6 +59,51 @@ class WebService {
     }
 }
 
+// MARK: UserAgent
+typealias LibraryVersion = (name: String, version: String)
+extension WebService {
+  
+  /// The operating system's name.
+  ///
+  /// - returns: The operating system's name, or nil if it could not be determined.
+  ///
+  internal static func osName() -> String? {
+    #if os(iOS)
+    return "iOS"
+    #elseif os(OSX)
+    return "macOS"
+    #elseif os(tvOS)
+    return "tvOS"
+    #elseif os(watchOS)
+    return "watchOS"
+    #else
+    return nil
+    #endif
+  }
+  
+  internal static func computeUserAgent() -> String {
```

Comment:
I would add a small comment stating that it will look something like `insights-ios (2.1.0); iOS (12.1.0)`